connectors/uplynk/src/main/java/com/theoplayer/android/connector/uplynk/UplynkConnector.kt

```diff
@@ -0,0 +1,48 @@
+package com.theoplayer.android.connector.uplynk
+
+import com.theoplayer.android.api.THEOplayerView
+import com.theoplayer.android.api.ads.ServerSideAdIntegrationController
+import com.theoplayer.android.api.source.ssai.CustomSsaiDescriptionRegistry
+import com.theoplayer.android.connector.uplynk.common.EventDispatcherImpl
+
+internal const val TAG = "UplynkConnector"
+
+/**
+ * A connector for the Uplynk Media Platform.
+ *
+ * @param theoplayerView
+ *   The THEOplayer view, which will be connected to the created connector.
+ */
+class UplynkConnector(
+    private val theoplayerView: THEOplayerView,
+) {
+    private lateinit var integration: UplynkAdIntegration
+    val eventDispatcher = EventDispatcherImpl()
```

Comment:
Use a dedicated listener interface for Uplynk, instead of fiddling with `EventDispatcher` and all this `EventType` madness.

See [YospaceConnector.addListener()](https://github.com/THEOplayer/android-connector/blob/7.9.0/connectors/yospace/src/main/java/com/theoplayer/android/connector/yospace/YospaceConnector.kt#L70-L82) and [YospaceListener](https://github.com/THEOplayer/android-connector/blob/7.9.0/connectors/yospace/src/main/java/com/theoplayer/android/connector/yospace/YospaceListener.kt) for an example.

Reply:
Do you propose to have listeners instead of events? 

We have this `EventType` stuff in our main API. It's not ideal but I think it worth to be consistent in the connector. Or am I misunderstanding?

Reply:
Yes, I suggest we get rid of the `Event` types for this connector, in favor of a single `Listener` interface with a method for each "event".

We're already doing this in the Yospace connector. And to be honest, we should start doing it in the THEOplayer SDK too. 😛

Reply:
Hah) I just learned wtf and whytf events have this signature `Event<E extends Event>` and now your sayin we should forget about it??

Reply:
I made the change. Now if you want to get Uplynk response data you have to add listener as here:
```
       uplynkConnector.addListener(object: UplynkListener {
            override fun onPreplayResponse(response: PreplayResponse) {
                Log.d("UplynkConnectorEvents", "PREPLAY_RESPONSE $response")
            }

            override fun onAssetInfoResponse(response: AssetInfoResponse) {
                Log.d("UplynkConnectorEvents", "ASSET_INFO_RESPONSE $response")
            }

        })

        theoplayerView.player.ads.addEventListener(AdsEventTypes.AD_ERROR) {
            Log.d("UplynkConnectorEvents", "AD_ERROR " + it.error)
        }
```

And now I think that it's a bit awkward that clients should have `onAssetInfoResponse` to process successful responses for AssetInfo requests and to process errors they should use `theoplayerView.player.ads.addEventListener(AdsEventTypes.AD_ERROR)`. 

I propose to have `onAssetInfoFailure` listener in case asset info response fails. What do you think?

Reply:
We can always do both: call an integration-specific listener method *and* dispatch a generic `AdErrorEvent`.

Reply:
I made it: 
```
      uplynkConnector.addListener(object: UplynkListener {
            override fun onPreplayResponse(response: PreplayResponse) {
                Log.d("UplynkConnectorEvents", "PREPLAY_RESPONSE $response")
            }

            override fun onAssetInfoResponse(response: AssetInfoResponse) {
                Log.d("UplynkConnectorEvents", "ASSET_INFO_RESPONSE $response")
            }

            override fun onPreplayFailure(exception: Exception) {
                Log.d("UplynkConnectorEvents", "PREPLAY_RESPONSE_FAILURE $exception")
            }

            override fun onAssetInfoFailure(exception: Exception) {
                Log.d("UplynkConnectorEvents", "ASSET_INFO_RESPONSE Failure $exception")
            }

        })

        theoplayerView.player.ads.addEventListener(AdsEventTypes.AD_ERROR) {
            Log.d("UplynkConnectorEvents", "AD_ERROR " + it.error)
        }

```